### PR TITLE
Set bootstrap env vars based on Cassandra resources

### PR DESCRIPTION
**Pull request by cscetbon**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
This PR fixes a bug where the env CASSANDRA_MAX_HEAP is set based on that container resources instead of the main Cassandra container's resources.

### CODE REVIEW

File: pkg/controller/cassandracluster/generator.go [hunk 1]
```diff
@@ -395,7 +401,7 @@ func generateCassandraStatefulSet(cc *api.CassandraCluster, status *api.Cassandr
 	// Merge cassandra main container environment variables into sidecars.
 	for idx, container := range ss.Spec.Template.Spec.Containers {
 		if container.Name != cassandraContainerName {
-			ss.Spec.Template.Spec.Containers[idx].Env = append(ss.Spec.Template.Spec.Containers[idx].Env, cassandraContainer.Env...)
+			ss.Spec.Template.Spec.Containers[idx].Env = append(container.Env, bootstrapContainer.Env...)
```
Comment (cscetbon) [inline pkg/controller/cassandracluster/generator.go:404]: @erdrix , do you think we really need it ? We could probably remove some of them and maybe push only DC, RACK and CLUSTER_NAME ? why would they need the heap size, the seeds etc.. ?

Comment (erdrix) [reply pkg/controller/cassandracluster/generator.go:406]: When I did it, I took the side of pushing them all because we had a low visibility on sidecar usage and therefore the env vars required for these ... So if the necessary are the ones you specified no problems for me :)

File: pkg/controller/cassandracluster/generator.go [hunk 2]
```diff
@@ -577,33 +585,15 @@ func createEnvVarForCassandraContainer(cc *api.CassandraCluster, status *api.Cas
 				},
 			},
 		},
-		v1.EnvVar{
-			Name: "POD_NAME",
```
Comment (erdrix) [inline pkg/controller/cassandracluster/generator.go:581]: Did you remove this one because it's unused ?

File: pkg/controller/cassandracluster/generator.go [hunk 3]
```diff
@@ -577,33 +585,15 @@ func createEnvVarForCassandraContainer(cc *api.CassandraCluster, status *api.Cas
...
 			Name:  "CASSANDRA_GC_STDOUT",
 			Value: strconv.FormatBool(cc.Spec.GCStdout),
 		},
-		v1.EnvVar{
+		{
 			Name:  "CASSANDRA_NUM_TOKENS",
 			Value: strconv.Itoa(int(numTokensPerRacks)),
 		},
-		v1.EnvVar{
-			Name: "NODE_NAME",
```
Comment (erdrix) [inline pkg/controller/cassandracluster/generator.go:598]: Did you remove this one because it's unused ?

File: pkg/controller/cassandracluster/generator.go [hunk 4]
```diff
@@ -750,7 +739,17 @@ func createCassandraContainer(cc *api.CassandraCluster, status *api.CassandraClu
 				},
 			},
 		},
-		Env: createEnvVarForCassandraContainer(cc, status, resources, dcRackName),
```
Comment (erdrix) [inline pkg/controller/cassandracluster/generator.go:753]: There is no need of all the env vars that were push here before ? (Any ideas why they were pushed before ? Just to be sure that we don't miss anything else like I did for the MAX_HEAP :D )

File: test/e2e/cassandracluster_test.go [hunk 5]
```diff
@@ -405,8 +405,8 @@ func cassandraClusterCleanupTest(t *testing.T, f *framework.Framework, ctx *fram
 			return false, nil
 		}
 
-		if !endTime.After(startTime) {
-			t.Logf("Expected endTime (%s) to be after startTime (%s)", endTime, startTime)
+		if endTime.Sub(startTime) < 0 {
```
Comment (erdrix) [inline test/e2e/cassandracluster_test.go:408]: Why did you change this test ?

File: pkg/controller/cassandracluster/generator.go (hunk 2 above)
Comment (cscetbon) [reply pkg/controller/cassandracluster/generator.go:581]: Correct

File: pkg/controller/cassandracluster/generator.go (hunk 3 above)
Comment (cscetbon) [reply pkg/controller/cassandracluster/generator.go:598]: yes

File: pkg/controller/cassandracluster/generator.go (hunk 4 above)
Comment (cscetbon) [reply pkg/controller/cassandracluster/generator.go:753]: the Cassandra container does not need them

File: test/e2e/cassandracluster_test.go (hunk 5 above)
Comment (cscetbon) [reply test/e2e/cassandracluster_test.go:408]: because some time it runs even in the same second so startTime = endTime and the test fails.
